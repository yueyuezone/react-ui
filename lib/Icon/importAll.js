function importAll(r) {
    r.keys().forEach(r)
}
try {
    importAll(require.context('./iconSvg/',true,/\.svg$/))
} catch (e) {
    //console.log(e)
}
