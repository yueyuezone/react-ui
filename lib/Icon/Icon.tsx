import React, {SVGAttributes} from 'react';
import './importAll';
import './icon.scss';
import { cn } from '../util/util'

interface IconProps extends React.SVGAttributes<SVGElement> {
    name: string
}


const Icon: React.FunctionComponent<IconProps> = (props) => {
    const {className,name, ...restprops} = props;
    return (
        <svg className={cn('lufei-icon',className)} {...restprops}>
            <use xlinkHref={`#${name}`}></use>
        </svg>
    );
};
export default Icon;