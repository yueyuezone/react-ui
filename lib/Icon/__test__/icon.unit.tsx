import React from 'react'
import Icon from '../Icon'
import renderer from 'react-test-renderer'
import {mount} from "enzyme"

describe('icon测试',()=>{
    it('测试UI组件',()=>{
        //console.log(Icon)
        const iconCom = renderer.create(<Icon name="file" className="ui-icon"></Icon>)
        let iconJson = iconCom.toJSON()
        expect(iconJson).toMatchSnapshot('icon')
    })
    it('测试onclick⌚️',  ()=> {
        const fn = jest.fn()
        const iconCom = mount(<Icon name="bank" onClick={fn}></Icon>)
        iconCom.find('svg').simulate('click')
        expect(fn).toBeCalled()
    });
})