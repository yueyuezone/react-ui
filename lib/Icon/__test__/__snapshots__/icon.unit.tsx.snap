// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon测试 测试UI组件: icon 1`] = `
<svg
  className="lufei-icon ui-icon"
>
  <use
    xlinkHref="#file"
  />
</svg>
`;
